app/auth/jupyterhub_auth.py

```diff
@@ -16,41 +16,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Implement JupyterHub authentication workflow."""
-
-import json
 import re
-from urllib.parse import parse_qs, urlencode, urljoin
+from urllib.parse import urlencode, urljoin
 
-import jwt
 import requests
-from oic import rndstr
 from flask import Blueprint, current_app, redirect, request, Response, session, url_for
 
-from .web import JWT_ALGORITHM, get_key_for_user
+from .oauth_provider_app import JupyterHubProviderApp
+from .utils import (
+    get_redis_key_from_session,
+    get_redis_key_from_token,
+    handle_login_request,
+    handle_token_request,
+)
 
+JH_SUFFIX = "jh_oauth_client"
 
 blueprint = Blueprint("jupyterhub_auth", __name__, url_prefix="/auth/jupyterhub")
 
 
 class JupyterhubUserToken:
     def process(self, request, headers):
```

Comment:
Can we document the interface here? For example `request` is not being used, but I would assume that it needs to be here to comply with usage interface - adding a short description would help a lot.

Reply:
See #232